src/Plugins/RestServer/Helpers/ContractHelper.cs

```diff
@@ -0,0 +1,181 @@
+        {
+            var contractState = NativeContract.ContractManagement.GetContract(snapshot, scriptHash);
+            if (contractState == null)
+                return false;
+            return IsNep17Supported(contractState);
+        }
+
+        public static bool IsNep11Supported(DataCache snapshot, UInt160 scriptHash)
+        {
+            var contractState = NativeContract.ContractManagement.GetContract(snapshot, scriptHash);
+            if (contractState == null)
+                return false;
+            return IsNep11Supported(contractState);
+        }
+
+        public static bool IsNep17Supported(ContractState contractState)
+        {
+            var manifest = contractState.Manifest;
+            if (manifest.SupportedStandards.Any(a => a.Equals("NEP-17")))
+            {
+                try
+                {
+                    var symbolMethod = manifest.Abi.GetMethod("symbol", 0);
+                    var decimalsMethod = manifest.Abi.GetMethod("decimals", 0);
+                    var totalSupplyMethod = manifest.Abi.GetMethod("totalSupply", 0);
+                    var balanceOfMethod = manifest.Abi.GetMethod("balanceOf", 1);
+                    var transferMethod = manifest.Abi.GetMethod("transfer", 4);
+
+                    var symbolValid = symbolMethod.Safe == true &&
+                        symbolMethod.ReturnType == ContractParameterType.String;
+                    var decimalsValid = decimalsMethod.Safe == true &&
+                        decimalsMethod.ReturnType == ContractParameterType.Integer;
+                    var totalSupplyValid = totalSupplyMethod.Safe == true &&
+                        totalSupplyMethod.ReturnType == ContractParameterType.Integer;
+                    var balanceOfValid = balanceOfMethod.Safe == true &&
+                        balanceOfMethod.ReturnType == ContractParameterType.Integer &&
+                        balanceOfMethod.Parameters[0].Type == ContractParameterType.Hash160;
+                    var transferValid = transferMethod.Safe == false &&
+                        transferMethod.ReturnType == ContractParameterType.Boolean &&
+                        transferMethod.Parameters[0].Type == ContractParameterType.Hash160 &&
+                        transferMethod.Parameters[1].Type == ContractParameterType.Hash160 &&
+                        transferMethod.Parameters[2].Type == ContractParameterType.Integer &&
+                        transferMethod.Parameters[3].Type == ContractParameterType.Any;
+                    var transferEvent = manifest.Abi.Events.Any(s =>
+                        s.Name == "Transfer" &&
+                        s.Parameters.Length == 3 &&
+                        s.Parameters[0].Type == ContractParameterType.Hash160 &&
+                        s.Parameters[1].Type == ContractParameterType.Hash160 &&
+                        s.Parameters[2].Type == ContractParameterType.Integer);
```

Comment:
We have to remove these safe method check, because safe doesn't exist at the beginning. Old contract hasn't supported safe label unless force them to update. It's also not forced in protocol. https://github.com/neo-project/proposals/blob/master/nep-17.mediawiki